### PR TITLE
Hotgrid updates 2017-11-17

### PR DESCRIPTION
**Pull request by boonier**
Refactors the notify event to off load the rendering to the template
Tweaks to the CSS states (if < 1 image)
Add _hasRoundedCorners flag to be set in component json

### CODE REVIEW

File: properties.schema [hunk 1]
```diff
@@ -38,6 +38,14 @@
       "validators": ["number"],
       "help": "This value determines the number of columns within the grid. Any number of columns can be set however keep in mind the more columns there are the smaller the items will be."
     },
+    "_hasRoundedCorners": {
+      "type":"boolean",
+      "required":false,
+      "inputType": "Boolean",
```
Comment (moloko) [inline properties.schema]: Could you change this to `"inputType": "Checkbox"`?

File: less/hotgrid.less [hunk 2]
```diff
@@ -117,7 +122,7 @@
 				width: 100%;
 				background-color: fade(@item-color-disabled, 50%);
 				/* uncomment if circular items */
-				//border-radius: 50%; 
+				border-radius: 50%;
```
Comment (moloko) [inline less/hotgrid.less]: should this line be included? isn't it now handled by lines 38 - 41? either way the comment should be removed

File: less/hotgrid.less [hunk 3]
```diff
@@ -97,6 +101,7 @@
 .hotgrid-component {
 	.hotgrid-item-states-css {
 		border: 3px solid transparent;
+
 		/* uncomment if circular items */
 		// border-radius: 50%;
```
Comment (moloko) [inline less/hotgrid.less]: is this needed now?

**Comment by kirsty-hames** [comment]
Do you think we could add a '_classes' attribute to the items? I often have issues where content quantity varies across items and some popups would benefit from a different layout. Would be nice to be able to set classes for different layouts e.g 70/30 rather than 50/50 etc ?

File: less/hotgrid.less (hunk 2 above)
Comment (boonier) [reply less/hotgrid.less]: Yep it's not needed now, will delete

**Comment by boonier** [comment]
@moloko I'm not sure, @guywillis or @kirsty-hames does Matt's comment above apply to the changes I've made to adapt-hotgrid.js?

**Comment by boonier** [comment]
@kirsty-hames re adding layout classes to the notify popups, is that something that I can work on, or does it need to go to committee?

**Comment by boonier** [comment]
@moloko The changes I made don't require the latest - it works with the current version of notify

**Pull request merged**